### ISSUE TITLE
12$: improve error recovery for missing 'function' keyword after 'async'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1012]
```javascript
async parsePDBStreamDirectoryAsync(reader, superBlock) {                         
  return await withLoadScopeAsync(() => { 
  });
}
```
The keyword `function` is missing between `async` and `parsePDBStreamDirectoryAsync`. quick-lint-js reports an error on the `(` ("E0176: missing arrow operator for arrow function"), but it should instead report that `function` is missing after `async`.

**Comment by jaitjacob** [comment]
for the given sample code current implementation raises following errors,
1. error: missing arrow operator for arrow function [E0176]
2. error: missing arrow operator for arrow function [E0176] _---------> (reported twice)_
3. error: 'await' is only allowed in async functions [E0162]
4. warning: use of undeclared variable: withLoadScopeAsync [E0057]

@strager do you want the fix to raise a new error stating 'function' is missing?
E0162 from above seems appropriate foo what we want here instead of a new diag?

**Comment by strager** [comment]
> @strager do you want the fix to raise a new error stating 'function' is missing?

Yes. I don't see any existing diagnostic type for this issue, so we'll need to make a new one.

> E0162 from above seems appropriate foo what we want here instead of a new diag?

E0162 should not be reported because quick-lint-js should know that the user intended to make an async function.
